src/node/channels.h

```diff
@@ -597,31 +589,34 @@ namespace ccf
         kex_ctx.reset();
       }
 
-      kex_ctx.load_peer_key_share(ks);
+      kex_ctx.load_peer_key_share(peer_ks);
 
       if (status != ESTABLISHED)
         status = WAITING_FOR_FINAL;
 
       // We are the responder and we return a signature over both public key
       // shares back to the initiator
 
-      auto oks = kex_ctx.get_own_key_share();
-      std::vector<uint8_t> pks = {ks.p, ks.p + ks.n};
-      auto serialised_signed_share = sign_key_share(oks, false, &pks);
+      auto space = 1 * sizeof(size_t);
+      std::vector<uint8_t> payload(space);
+      auto data_ = payload.data();
+      serialized::write(data_, space, ChannelMsg::key_exchange_response);
+
+      sign_key_share(payload, kex_ctx.get_own_key_share(), false, peer_ks);
```

Comment:
I changed the API of `sign_key_share` to serialise into an out-vector before I added the non-contiguous-ByteArray change for non-key-exchange messages. So this still creates one mega-payload (as it did previously), but could be improved to just use the existing buffers in-place. I'll do that in a future PR.